### ISSUE TITLE
Checkbox Infinite Loop

### ISSUE THREAD

**Issue by ReneVallecillo** [angular/material2#812]
#### Bug, feature request, or proposal:

Bug
#### What is the expected behavior?
Setting 

```
<li *ngFor="let item of data">
    <md-checkbox [checked]="item.checked" (change)="item.checked=!item.checked"> 
    {{item.name}}
    </md-checkbox>
```
Should change the source value and update data once on _click/change_

#### What is the current behavior?

The checkbox goes on an infinite switching loop

#### What are the steps to reproduce?

http://plnkr.co/edit/o7vPsr7S7L14CtQuiDOP?p=preview

Simpler version:

https://plnkr.co/edit/OhqPeWHnjfvkxyNObETc?p=preview

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: http://plnkr.co/edit/o077B6uEiiIgkC0S06dd

**Comment by mmrath** [comment]
I think this is related to https://github.com/angular/material2/issues/575 

I believe there is WIP to make md-check-box behave like native checkbox and fire onchange only if the control is in focus. That probably should resolve this

**Comment by ReneVallecillo** [comment]
@mmrath I guess they are related at some level. But I checked your plnkr and I didn't see the infinite loop. 

The on change on focus will be welcome, since in theory will allow me to keep two controls binded to one data source. As of right now, since both react on change, they just keep calling each others change. Or so it seems.

**Comment by ReneVallecillo** [comment]
#582

**Comment by jelbourn** [comment]
@ReneVallecillo What version of angular material were you using?

**Comment by ReneVallecillo** [comment]
@jelbourn The infinite loop shows on the plunker, which is using the suggested template. On my repo, I'm using version **alpha6** something different but related may be happening.

What is happening in my repo is, since I have two components bound to a single data source, the change is triggering in both components and that seems to trigger @ #575 as:

1. Checkbox component 1 is checked, change event is triggered
2. Data source is updated
3. Checkbox component 2 is checked, but change event is also triggered
4. ???
5. Endless Loop.

Or so it seems. This should be fixed with #791. But in my repo what it shows is https://github.com/angular/angular/issues/6005#issuecomment-230395099

I think this prevents the endless loop to show on my repo. 

But I don't understand why it happens here: 

https://plnkr.co/edit/OhqPeWHnjfvkxyNObETc?p=preview

It is a really basic code. 

TLDR, I'm on alpha 6, but while replicating my code on plunker, the endless loop happens.

**Comment by jelbourn** [comment]
I think this will be fixed by #820.

**Comment by DevVersion** [comment]
@jelbourn Jut confirmed with #820, and it seems to fix that issue.

It's just important (again) to make the adjustments with the emitted `MdSlideToggleChange` class.
```html
<md-checkbox ... (change)="item.checked = $event.checked">
```
